lib/src/config_handler.dart

```diff
@@ -102,6 +102,43 @@ class ConfigHandler {
     configFileLastModified = configFile.lastModifiedSync();
   }
 
+  /// Will increment the version number and update the date
+  /// in the config file for the provided tool name while
+  /// also incrementing the version number in [ToolInfo]
+  void incrementToolVersion({required String tool}) {
+    if (!parsedTools.containsKey(tool)) {
+      return;
+    }
+
+    // Read in the config file contents and use a regex pattern to
+    // match the line for the current tool (ie. flutter-tools=2023-01-05,1)
+    final String configString = configFile.readAsStringSync();
+    final String pattern = '^($tool)=([0-9]{4}-[0-9]{2}-[0-9]{2}),([0-9]+)\$';
+
+    final RegExp regex = RegExp(pattern, multiLine: true);
+    final Iterable<RegExpMatch> matches = regex.allMatches(configString);
+
+    // TODO: need to determine what to do when there are two lines for the same tool
+    //  as outlined in the Dev Notes document; currently only assuming one line per tool
+    if (matches.length == 1) {
```

Comment:
Could we raise an exception as not implemented when `matches.length > 1` and a test for that?

Reply:
Great idea, I actually should just implement the method that will get triggered when the matches aren't equal to 1 so I can get rid of this TODO

Reply:
Just added the functionality to reset the configuration file if that conditional fails. Also added tests for it